### PR TITLE
Unit-test framework to work with MetricStore

### PR DESCRIPTION
**Pull request by abaranau**
Follow up on: https://github.com/caskdata/cdap/pull/1251

Addresses:

* unit-test framework

### CODE REVIEW

File: cdap-app-fabric/src/main/java/co/cask/cdap/test/RuntimeStats.java [hunk 1]
```diff
@@ -154,37 +157,32 @@ private void doWaitFor(String name, long count, long timeout, TimeUnit timeoutUn
...
+  private static long getTotalCounter(Map<String, String> context, String metricName) {
+    CubeQuery query = getTotalCounterQuery(context, metricName);
+    try {
+      Collection<TimeSeries> result = metricStore.query(query);
+      if (result.isEmpty()) {
+        return 0;
       }
-    });
+      // since it is totals query and not groupBy specified, we know there's one time series with one time value
+      return result.iterator().next().getTimeValues().get(0).getValue();
```
Comment (gsps1) [inline cdap-app-fabric/src/main/java/co/cask/cdap/test/RuntimeStats.java]: do we need a hasNext check before this accessing the time-series ? if the data is not yet collected we may not get a time series value yet and we we want to retry before throwing exception ?

**Comment by gsps1** [comment]
just one comment, LGTM otherwise :+1:

File: cdap-app-fabric/src/main/java/co/cask/cdap/test/RuntimeStats.java (hunk 1 above)
Comment (abaranau) [inline cdap-app-fabric/src/main/java/co/cask/cdap/test/RuntimeStats.java]: in that case we will not have time series at all...

Comment (abaranau) [inline cdap-app-fabric/src/main/java/co/cask/cdap/test/RuntimeStats.java]: let me add a check just in case though :)

**Pull request merged**